api/python/quilt3/data_transfer.py

```diff
@@ -22,29 +23,145 @@
 from .util import PhysicalKey, QuiltException
 
 
-def create_s3_client():
-    botocore_session = create_botocore_session()
-    boto_session = boto3.Session(botocore_session=botocore_session)
 
-    # Check whether credentials are present
-    if boto_session.get_credentials() is None:
-        # Use unsigned boto if credentials aren't present
-        s3_client = boto_session.client('s3', config=Config(signature_version=UNSIGNED))
-    else:
-        # Use normal boto
-        s3_client = boto_session.client('s3')
 
-    # Enable/disable file read callbacks when uploading files.
-    # Copied from https://github.com/boto/s3transfer/blob/develop/s3transfer/manager.py#L501
-    event_name = 'request-created.s3'
-    s3_client.meta.events.register_first(
-        event_name, signal_not_transferring,
-        unique_id='datatransfer-not-transferring')
-    s3_client.meta.events.register_last(
-        event_name, signal_transferring,
-        unique_id='datatransfer-transferring')
 
-    return s3_client
+class S3Api(Enum):
+    GET_OBJECT = "GET_OBJECT"
+    HEAD_OBJECT = "HEAD_OBJECT"
```

Comment:
There's no point making a distinction between `HEAD` and `GET` - it's a single S3 permission.

But even `GET` vs `LIST` is tricky - see the 403/404 comment.